<!--
Provides a reasonable set of parameters for running Karto on a robot
Assumes an 'odom_combined' frame, that provides reasonably accurate odometry
-->

<launch>
  <node pkg="slam_karto" type="slam_karto" name="slam_karto" output="screen">
    <remap from="scan" to="base_scan"/>
    <param name="odom_frame" value="odom"/>
    <param name="map_update_interval" value="25"/>
    <param name="resolution" value="0.025"/>
    <param name="use_scan_matching" value="true"/>
    <param name="use_scan_barycenter" value="true"/>
    <param name="minimum_travel_distance" value="0.2"/>
    <param name="minimum_travel_heading" value="0.174532925199433"/>
    <param name="scan_buffer_size" value="70"/>
    <param name="scan_buffer_max_scan_distance" value="20"/>
    <param name="link_match_min_response_fine" value="0.8"/>
    <param name="link_scan_max_distance" value="10"/>
    <param name="loop_search_max_distance" value="4.0"/>
    <param name="do_loop_closing" value="true"/>
    <param name="loop_match_min_chain_size" value="10"/>
    <param name="loop_match_max_variance_coarse" value="0.16"/>
    <param name="loop_match_min_response_coarse" value="0.8"/>
    <param name="loop_match_min_response_fine" value="0.8"/>
    <param name="corr_search_space_dim" value="0.3"/>
    <param name="corr_search_space_res" value="0.01"/>
    <param name="corr_search_space_smear_dev" value="0.03"/>
    <param name="loop_search_space_dim" value="8"/>
    <param name="loop_search_space_res" value="0.05"/>
    <param name="loop_search_space_smear_dev" value="0.03"/>
    <param name="distance_variance_penalty" value="0.09"/>
    <param name="angle_variance_penalty" value="0.121846967914683"/>
    <param name="fine_search_angle_offset" value="0.00349065850398866"/>
    <param name="coarse_search_angle_offset" value="0.349065850398866"/>
    <param name="coarse_angle_resolution" value="0.0349065850398866"/>
    <param name="minimum_angle_penalty" value="0.9"/>
    <param name="minimum_dist_penalty" value="0.5"/>
    <param name="use_response_expansion" value="false"/>

  </node>
</launch>
